server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -101,16 +107,24 @@ public class LocalStorageManager extends SingleStorageManager {
     try {
       countDownLatch.await();
     } catch (InterruptedException e) {
-      e.printStackTrace();
+      LOG.error("Failed to wait initializing local storage.", e);
     }
-
     executorService.shutdown();
+
     int failedCount = storageBasePaths.length - successCount.get();
-    if (failedCount > 0) {
-      throw new RuntimeException(String.format("[%s] local storage init failed!", failedCount));
+    long maxFailedNumber = conf.getLong(LOCAL_STORAGE_INITIALIZE_MAX_FAIL_NUMBER);
+    if (failedCount > maxFailedNumber || successCount.get() == 0) {
+      throw new RuntimeException(
+          String.format("Initialize %s local storage(s) failed, "
+              + "specified local storage paths size: %s, the conf of %s size: %s",
+              failedCount, localStorageArray.length, LOCAL_STORAGE_INITIALIZE_MAX_FAIL_NUMBER.key(), maxFailedNumber)
+      );
     }
-
-    localStorages = Arrays.asList(localStorageArray);
+    localStorages = Arrays.stream(localStorageArray).filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
Fix the bug introduced by #72 .

Reply:
For #72, it's not a bug.

Reply:
The localStorageArray maybe exist the null in some index, this will make the localStorages exist the null element when using the `Arrays.asList(localStorageArray)`

Reply:
Oh, my fault. Ignore this